cvmfs/sync_mediator.cc

```diff
@@ -775,47 +784,98 @@ void SyncMediator::CreateNestedCatalog(SharedPtr<SyncItem> directory) {
 
 void SyncMediator::RemoveNestedCatalog(SharedPtr<SyncItem> directory) {
   const std::string notice =  "Nested catalog at " + directory->GetUnionPath();
-  PrintChangesetNotice(kRemoveCatalog, notice);
+  reporter_->OnRemove(notice, catalog::DirectoryEntry());
 
   if (!params_->dry_run) {
     catalog_manager_->RemoveNestedCatalog(directory->GetRelativePath());
   }
 }
 
+void SyncDiffReporter::OnInit(const history::History::Tag &from_tag,
+                              const history::History::Tag &to_tag) {}
 
-void SyncMediator::PrintChangesetNotice(const ChangesetAction  action,
-                                        const std::string     &extra) const {
-  if (!params_->print_changeset) {
-    ++changed_items_;
-    if (changed_items_ % processing_dot_interval == 0) {
-      LogCvmfs(kLogPublish, kLogStdout | kLogNoLinebreak, ".");
-    }
-    return;
+void SyncDiffReporter::OnStats(const catalog::DeltaCounters &delta) {}
+
+void SyncDiffReporter::OnAdd(const std::string &path,
+                             const catalog::DirectoryEntry &entry) {
+  changed_items_++;
+  InternalAdd(path);
+}
+void SyncDiffReporter::OnRemove(const std::string &path,
+                                const catalog::DirectoryEntry &entry) {
+  changed_items_++;
+  InternalRemove(path, entry);
+}
+void SyncDiffReporter::OnModify(const std::string &path,
+                                const catalog::DirectoryEntry &entry_from,
+                                const catalog::DirectoryEntry &entry_to) {
+  changed_items_++;
+  InternalModify(path, entry_from, entry_to);
+}
+
+void SyncDiffReporter::PrintDots() {
+  if (changed_items_ % processing_dot_interval_ == 0) {
+    LogCvmfs(kLogPublish, kLogStdout | kLogNoLinebreak, ".");
   }
+}
+
+void SyncDiffReporter::InternalAdd(const std::string &path) {
+  const std::string keyword = "catalog";
+  const char *action_label;
 
-  const char *action_label = NULL;
-  switch (action) {
-    case kAdd:
-    case kAddCatalog:
-    case kAddHardlinks:
+  if (print_action_ == kPrintChanges) {
+    if (path.find(keyword) != std::string::npos) {
```

Comment:
That would also trigger on paths that happen to contain the word catalog.  I would suggest instead checking if the first character of `path` is `/`.  Before, we can assert that path is not empty.